### DISCUSSION TITLE
After recompiling the Docker file, an error occurred during execution.

### DISCUSSION THREAD

**Discussion by jianghu52** [jlesage/docker-firefox#253] [Q&A] [answered]
I downloaded the entire file from GitHub, recompiled it, and executed it on WSL, but the image encountered an error.
![Snipaste_2024-08-31_15-44-16](https://github.com/user-attachments/assets/2f59064f-f908-4d78-a2d4-f54e2b89094c)
the wrong detail is [[supervisor  ] could not load service 'app': could not load 'sgid': unknown group '#!/bin/sh'].

It's strange that I have no issues when I use the jlesage/firefox image directly. What steps might I be missing in docker building? By the way, the reason I need to recompile is that the current jlesage/firefox image lacks an encoder, which prevents me from loading videos on some video websites. So, I wanted to compile a Dockerfile myself that includes a decoder. Unfortunately, I've failed at just the first step, which is compiling the most basic Dockerfile. I'm feeling a bit pessimistic about the future.

**Comment by jlesage** [comment, marked as answer]
Looks like the file permissions didn't follow: some files have lost their execute permission with your build.

> the current jlesage/firefox image lacks an encoder, which prevents me from loading videos on some video websites

If you have an example to share, I can add the missing packages to the container image. Also, instead of re-compiling the image, you can use the `INSTALL_PACKAGES` environment variable to automatically install additional packages.

**Comment by jianghu52** [comment]
sorry, the video url I saw falls under the category of internal company training and cannot be disclosed. I think I should try using INSTALL_PACKAGES. Thank you very much for your response

**Comment by marcogruhl** [comment]
For future visitors:
when editing files on windows systems the line ending is eminent:
Windows CR+LF
Unix LF
Mac CR
e.g. in Notepad++ you find this under Edit -> EOL Conversion
